src/Examine.Core/Search/IQuery.cs

```diff
@@ -134,5 +135,38 @@ public interface IQuery
         /// <param name="maxInclusive"></param>
         /// <returns></returns>
         IBooleanOperation RangeQuery<T>(string[] fields, T? min, T? max, bool minInclusive = true, bool maxInclusive = true) where T : struct;
+
+        /// <summary>
+        /// Add a facet string to the current query
+        /// </summary>
+        /// <param name="field"></param>
+        /// <returns></returns>
+        IFacetQueryField Facet(string field);
```

Comment:
Will it make sense to allow doing this in the API?

```
var query = searcher.CreateQuery("content")
                        .Field("nodeName", "sydney")
                        .And()
                        .Facet("nodeName")
                        .Or()
                        .Facet("test")
                        .Not()
                        .Facet("test2");
```

Some things to consider:

1. Can we enable facets for multiple fields in a single query?
2. Can we enable facets for multiple fields with different facet types in a single query?
3. I assume the boolean nature of the above doesn't make sense?

My presumption based on your implementation is that 1 and 2 work and is ok to do. But number 3 doesn't make sense from an API perspective since the implementation just adds facet fields.

I think faceting needs to be treated differently in a separate interface like how sorting works whereby you need to build up your query first and then you can sort.  It should probably be a similar concept for facets, possibly facets come before ordering or vice versa. Or maybe faceting is just part of IOrdering?

Reply:
I hadn't thought about the boolean use case and agree that a separate interface for the faceting would be ideal. I've now moved the facet operations from `IQuery` to a new interface `IFaceting`. I'm not really sure if there's a better way to structure it but I've added this interface to the `IOrdering` interface because it does make sense to do the faceting after the query. I've also added `IFacetAppending` (Best name I could come up with on the spot 😅 ) which contains the `And()` method for appending other operations in the `IOrdering` interface. This is needed because the facet operations has their own operations that can be applied to set options I.E `MaxCount(int count)`.

Just making the tests build I can also see that the API seems eaiser to read now.